### DISCUSSION TITLE
Event N seconds after file modified

### DISCUSSION THREAD

**Discussion by channingwalton** [Taitava/obsidian-shellcommands#287] [Ideas - done & released]
I would like to run a script when a file is modified, but doing so on every keystroke is a bit heavy handed.

So, how about triggering the event some time after the file is modified, which would reduce the amount of unnecessary work being done?

**Comment by Taitava** [comment]
Hi and thank you for your suggestion! 🙂

A nice coincidence, there has actually been a bit offtopic discussion today in another event related thread about implementing some kind of threshold: https://github.com/Taitava/obsidian-shellcommands/discussions/284#discussioncomment-4184046 .

Also, what comes to file editing, Obsidian uses some delay between modifying a note (e.g. key presses) and saving the file. I think it's around two seconds. So that should give already some threshold, but of course I understand if you want to increase the threshold even more.

**Comment by Taitava** [comment]
Just to note here, too. This feature was released in `0.22.0` earlier this year, and it's called _event debouncing_.
